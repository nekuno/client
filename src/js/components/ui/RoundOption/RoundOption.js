import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './RoundOption.scss';
import OptionImageService from '../../../services/OptionImageService';

export default class RoundOption extends Component {

    static propTypes = {
        picture: PropTypes.string.isRequired,
    };

    render() {
        const {picture} = this.props;

        const thumbnail = OptionImageService.getThumbnailRound(picture);

        return (
            <div className={styles.roundOption}>
                <img src={thumbnail}/>
            </div>
        );
    }
}