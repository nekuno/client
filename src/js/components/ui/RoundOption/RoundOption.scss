@import '../../../../scss/_partials/colors';

.round-option {
  background-color: #fff;
  border-radius: 50%;
  -moz-border-radius: 50%;
  -webkit-border-radius: 50%;
  display: inline-block;
  margin: 5px 10px 0 0;
  text-align: center;
  height: 100px;
  width: 100px;

  img {
    border-radius: 50%;
    -moz-border-radius: 50%;
    -webkit-border-radius: 50%;
    margin-top: 1px;
    height: 100px;
    width: 100px;
  }
}