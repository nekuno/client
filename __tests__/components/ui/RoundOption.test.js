import React from 'react';
import RoundOption from '../../../src/js/components/ui/RoundOption/RoundOption.js';
import styles from '../../../src/js/components/ui/RoundOption/RoundOption.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test RoundOption component', () => {
    const mockFn = jest.fn();
    const roundoption = shallow(<RoundOption onClickHandler={mockFn}>Lorem ipsum</RoundOption>);

    it('should be defined', () => {
        expect(roundoption).toBeDefined();
    });
    it('should render correctly', () => {
        expect(roundoption).toMatchSnapshot();
    });
    it('RoundOption click event', () => {
        roundoption.find(`.${styles.roundoption}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
