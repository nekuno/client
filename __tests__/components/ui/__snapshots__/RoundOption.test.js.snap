// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RoundOption component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RoundOption
    onClickHandler={[MockFunction]}
  >
    Lorem ipsum
  </RoundOption>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        src="https://imagine.pre.nekuno.com/index.php/media/cache/profile_option_small?url=undefined"
      />,
      "className": "roundOption",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "src": "https://imagine.pre.nekuno.com/index.php/media/cache/profile_option_small?url=undefined",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          src="https://imagine.pre.nekuno.com/index.php/media/cache/profile_option_small?url=undefined"
        />,
        "className": "roundOption",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "https://imagine.pre.nekuno.com/index.php/media/cache/profile_option_small?url=undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
